### PR TITLE
upgrade to .net 8

### PR DESCRIPTION
**Pull request by gpolcode**
Upgrade to .net 8

#52

### CODE REVIEW

**Comment by gpolcode** [comment]
Looks like on the latest image of "Visual Studio 2022" from AppVeyor .Net 8 is not yet available.

https://www.appveyor.com/docs/windows-images-software/#:~:text=6%20SDK%206.0.413-,.NET%207%20SDK,-7.0.400

Edit:
https://github.com/appveyor/build-images/issues/140

**Comment by olsh** [comment]
Hi @gpolcode

Thank you for your contribution. 

I'd suggest building the tool project against [all supported frameworks](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core).

Something like this.

```xml
<TargetFrameworks>net6.0;net7.0;net8.0</TargetFrameworks>
```

And yes, let's wait for the AppVeyor update.

File: src/DotNet.Consolidate.Tests/TestData/TestSolution/Directory.build.props [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <Project>
     <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
```
Comment (olsh) [inline src/DotNet.Consolidate.Tests/TestData/TestSolution/Directory.build.props]: There is no need to update the test solution. We do not build it anyway.

File: src/DotNet.Consolidate/DotNet.Consolidate.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (olsh) [inline src/DotNet.Consolidate/DotNet.Consolidate.csproj]: Let's keep the previous versions too.

**Review by olsh — CHANGES_REQUESTED** [review]

File: src/DotNet.Consolidate.Tests/TestData/TestSolution/Directory.build.props (hunk 1 above)
Comment (gpolcode) [reply src/DotNet.Consolidate.Tests/TestData/TestSolution/Directory.build.props]: Reverted this change

File: src/DotNet.Consolidate/DotNet.Consolidate.csproj (hunk 2 above)
Comment (gpolcode) [reply src/DotNet.Consolidate/DotNet.Consolidate.csproj]: I changed it to your recommendation

File: build/_build.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFrameworks>net6.0;net7.0;net8.0</TargetFrameworks>
```
Comment (olsh) [inline build/_build.csproj]: There is no need to specify multiple targets for the build project.
We can use just the latest version here.

File: src/DotNet.Consolidate.Tests/DotNet.Consolidate.Tests.csproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFrameworks>net6.0;net7.0;net8.0</TargetFrameworks>
```
Comment (olsh) [inline src/DotNet.Consolidate.Tests/DotNet.Consolidate.Tests.csproj]: The same here, the tool is pretty simple and doesn't have any specific requirements for frameworks. So we don't need to run the test against multiple frameworks. The latest one should be enough.

**Review by olsh — CHANGES_REQUESTED** [review]

File: build/_build.csproj (hunk 3 above)
Comment (gpolcode) [reply build/_build.csproj]: Changed it to only target .net 8

File: src/DotNet.Consolidate.Tests/DotNet.Consolidate.Tests.csproj (hunk 4 above)
Comment (gpolcode) [reply src/DotNet.Consolidate.Tests/DotNet.Consolidate.Tests.csproj]: Changed it to only target .net 8

**Comment by gpolcode** [comment]
@olsh The pipeline should work now. https://github.com/appveyor/build-images/issues/140 got closed

**Comment by olsh** [comment]
Thank you for your contribution! 👍🏻

**Pull request merged**